packages/sign-client/src/controllers/engine.ts

```diff
@@ -878,6 +891,43 @@ export class Engine extends IEngine {
     }
   };
 
+  private addRequestToQueue = (request: PendingRequestTypes.Struct) => {
+    this.requestQueue.requests.push(request);
+  };
+
+  private afterResponse = (params: EngineTypes.RespondParams) => {
+    this.deletePendingSessionRequest(params.response.id, { message: "fulfilled", code: 0 });
+    // intentionally delay the emitting of the next pending request a bit
+    setTimeout(() => {
+      this.requestQueue.state = REQUEST_QUEUE_STATES.idle;
+      this.processRequestQueue();
+    }, toMiliseconds(this.requestQueueDelay));
+  };
+
+  private processRequestQueue = async () => {
+    if (this.requestQueue.state === REQUEST_QUEUE_STATES.active) {
```

Comment:
Q: Trying to fully grok what we're doing here: this is it effectively creating a lock state where queue state remains active until the oldest pending request has been responded to?

Reply:
Yes, or expires, this way we ensure all requests are responded to